File: rule-indexer-python/indexing_rules/long_words_rule.py
from .base import IndexingRule
from logics.long_word_list_logic import LongWordListLogic

class LongWordsRule(IndexingRule):
    def __init__(self):
        self.logic = LongWordListLogic()
    
    def apply_rule(self, content):
        long_words = self.logic.apply(content)
        print("Words longer than 5 characters:")
        for word in long_words:
            print(word)

File: rule-indexer-go/rules/long_words_rule.go
package rules

import (
    "fmt"
    "rule-indexer-go/logics"
)

type LongWordsRule struct {
    logic *logics.LongWordListLogic
}

func NewLongWordsRule() *LongWordsRule {
    return &LongWordsRule{
        logic: logics.NewLongWordListLogic(),
    }
}

func (l *LongWordsRule) ApplyRule(content string) {
    longWords := l.logic.Apply(content)
    fmt.Println("Words longer than 5 characters:")
    for _, word := range longWords {
        fmt.Println(word)
    }
}
